### PR TITLE
Consolidate two Utilities::MPI::broadcast() functions.

### PR DESCRIPTION
**Pull request by drwells**
Now that we have `if constexpr` we can get rid of a bunch of `std::enable_if`s.

### CODE REVIEW

**Review by tjhei — APPROVED** [review]

File: include/deal.II/base/mpi.h [hunk 1]
```diff
@@ -1193,70 +1193,36 @@ namespace Utilities
...
-    broadcast(const MPI_Comm     comm,
-              const T           &object_to_send,
-              const unsigned int root_process = 0);
-
-    /**
-     * This function sends an object @p object_to_send from the process @p root_process
-     * to all other processes.
+     * that accepts arbitrary data types `T`. If `T` is an MPI type then this
+     * function immediately calls `MPI_Bcast`. Otherwise it will first use
+     * Utilities::pack() (which in turn uses `boost::serialize`, see in
```
Comment (bangerth) [inline include/deal.II/base/mpi.h:1202]:
```suggestion
     * Utilities::pack() (which in turn uses `boost::serialize`, see
```

File: include/deal.II/base/mpi.h [hunk 2]
```diff
@@ -1193,70 +1193,36 @@ namespace Utilities
...
-     * @note Be aware that this function is typically a lot more
-     * expensive than an `MPI_Bcast` because the function will use
-     * boost::serialization to (de)serialize, and execute a second
-     * `MPI_Bcast` to transmit the size before sending the data
-     * itself. On the other hand, if you have a single element of
-     * a data type `T` that is natively supported by MPI, then the
-     * compiler will choose another broadcast() overload that is efficient.
-     * If you have an array of such elements, you should use the other
-     * broadcast() function in this namespace that takes a pointer and a count
-     * argument.
```
Comment (bangerth) [inline include/deal.II/base/mpi.h:1213]: I think this might be nice to retain.

Comment (drwells) [reply include/deal.II/base/mpi.h:1213]: Good idea - I fixed a few references in the text so it makes sense in a unified function.

File: include/deal.II/base/mpi.h (hunk 1 above)
Comment (drwells) [reply include/deal.II/base/mpi.h]: I combined this with the former note (now a warning)

**Review by kronbichler — APPROVED** [review]

**Pull request merged**